packages/cli/src/verify.ts

```diff
@@ -0,0 +1,80 @@
+import { forge, getRpcUrl } from "@latticexyz/common/foundry";
+import { Hex, createWalletClient, getCreate2Address, http } from "viem";
+import { salt } from "./deploy/common";
+import { ensureDeployer } from "./deploy/ensureDeployer";
+import { privateKeyToAccount } from "viem/accounts";
+import { MUDError } from "@latticexyz/common/errors";
+import { getWorldFactoryContracts } from "./deploy/ensureWorldFactory";
+import { Contract } from "./deploy/ensureContract";
+
+type VerifyOptions = {
+  foundryProfile?: string;
+  systems: Contract[];
+  modules: Contract[];
+  worldAddress?: Hex;
+};
+
+async function verifyContract(
+  foundryProfile: string | undefined,
+  deployerAddress: Hex,
+  contract: Contract,
+  rpc: string,
+  cwd?: string,
+) {
+  if (!contract.label) {
+    throw new MUDError("Need a label");
+  }
+
+  const system = getCreate2Address({ from: deployerAddress, bytecode: contract.bytecode, salt });
+
+  await forge(["verify-contract", system, contract.label, "--verifier", "sourcify", "--rpc-url", rpc], {
```

Comment:
can we make the verifier here a cli argument so it also works with other verifiers than sourcify?

Reply:
also i guess we need the verifier url (either in the foundry toml or cli arg)